### PR TITLE
Decouple spacepoint formation from clusterization algorithm

### PR DESCRIPTION
**Pull request by beomki-yeo**
This PR takes out the `spacepoint_formation` from `clusterization_algorithm`.  

I don't like that `clusterzation_algorithm` has two outputs (yeah I did it...). Rigorously spacepoint formation is not a part of clusterization as well.

Considering `track_parameter_estimation` (global to local) is a standalone algorithm, I also wanted to keep the symmetry with `spacepoint_formation` (local to global)

Unfortunately I had to remove some openMP paralellizations, thus, @georgi-mania also needs to give a permission on this

### CODE REVIEW

**Review by krasznaa — APPROVED** [review]
Logically I'm on board with this. But as you absolutely correctly pointed out Beomki, we need to hear from @georgi-mania before we would potentially drastically worsen the OpenMP code's performance... :thinking:

File: examples/run/openmp/io_dec_par_example.cpp [hunk 1]
```diff
@@ -29,64 +28,36 @@ traccc::demonstrator_result run(traccc::demonstrator_input input_data,
...
+            Clusterization
+          -------------------*/
+
+        auto measurements_per_event = ca(cells_per_event);
+
+        /*------------------------
+            Spacepoint formation
+          ------------------------*/
+
+        auto spacepoints_per_event = sf(measurements_per_event);
```
Comment (georgi-mania) [inline examples/run/openmp/io_dec_par_example.cpp:61]: Here is missing the update for the counters (`n_modules, n_cells, n_measurements, n_spacepoints`) and a critical region to aggregate the measurements and spacepoints.

File: examples/run/openmp/par_example.cpp [hunk 2]
```diff
@@ -44,105 +43,52 @@ int par_run(const std::string &detector_file, const std::string &cells_dir,
...
+
+        /*----------------------------
+          Statistics
+          ----------------------------*/
+
+        n_modules += cells_per_event.size();
+        n_cells += cells_per_event.total_size();
+        n_measurements += measurements_per_event.total_size();
+        n_spacepoints += spacepoints_per_event.total_size();
     }
```
Comment (georgi-mania) [inline examples/run/openmp/par_example.cpp:84]: Here it should be a critical region to aggregate the measurements and spacepoints.

File: examples/run/openmp/par_example.cpp [hunk 3]
```diff
@@ -44,105 +43,52 @@ int par_run(const std::string &detector_file, const std::string &cells_dir,
...
```
Comment (georgi-mania) [inline examples/run/openmp/par_example.cpp:85]: In order to have a `par_example` with the same output as a `seq_example`, writing the .csv files with the measurements and spacepoints can be copied back here.

File: core/include/traccc/clusterization/clusterization_algorithm.hpp [hunk 4]
```diff
@@ -9,26 +9,18 @@
...
 #include "traccc/edm/measurement.hpp"
 #include "traccc/edm/spacepoint.hpp"
 #include "traccc/geometry/pixel_data.hpp"
 
 // clusterization
 #include "traccc/clusterization/component_connection.hpp"
 #include "traccc/clusterization/measurement_creation.hpp"
-#include "traccc/clusterization/spacepoint_formation.hpp"
-
-#ifdef _OPENMP
```
Comment (georgi-mania) [inline core/include/traccc/clusterization/clusterization_algorithm.hpp:22]: If we removed the OpenMP include from here, then each example from the `openmp` folder should include it.

File: examples/run/openmp/par_example.cpp (hunk 3 above)
Comment (beomki-yeo) [reply examples/run/openmp/par_example.cpp:85]: I will disable the write_measurement or write_spacepoint in #177, so let me skip it

File: examples/run/openmp/io_dec_par_example.cpp (hunk 1 above)
Comment (beomki-yeo) [reply examples/run/openmp/io_dec_par_example.cpp:65]: It's fixed now

File: examples/run/openmp/par_example.cpp (hunk 2 above)
Comment (beomki-yeo) [reply examples/run/openmp/par_example.cpp:86]: Done

File: core/include/traccc/clusterization/clusterization_algorithm.hpp (hunk 4 above)
Comment (beomki-yeo) [reply core/include/traccc/clusterization/clusterization_algorithm.hpp:22]: Yeah I included it in each example file

**Review by georgi-mania — APPROVED** [review]

**Comment by beomki-yeo** [comment]
I think somebody needs to renew the approval

**Review by stephenswat — APPROVED** [review]

**Pull request merged**